### PR TITLE
Fix SetComponentFlowRate for series active branch component sweeps

### PR DESCRIPTION
**Pull request by EnergyArchmage**
Pull request overview
---------------------
address issue #7125 related to central plant utility that sets and limits flow rates for series active branches that have multiple devices on a single branch.  Small coding mistake that may not make much of difference in practice. 

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add to input rules file for interfaces
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Required documentation updates?
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

**Comment by rraustad** [comment]
Similar unit test failure on Mac as for #7062. Not sure it's real but is now suspicious as to cause.

File: src/EnergyPlus/PlantUtilities.cc [hunk 1]
```diff
@@ -362,15 +362,16 @@ namespace PlantUtilities {
                     EMSLoadOverride = false;
 
                     for (CompNum = 1; CompNum <= loop_side.Branch(BranchIndex).TotalComponents; ++CompNum) {
-                        CompInletNodeNum = comp.NodeNumIn;
+                        auto &thisComp(loop_side.Branch(BranchIndex).Comp(CompNum));
+                        CompInletNodeNum = thisComp.NodeNumIn;
```
Comment (Myoldmopar) [inline src/EnergyPlus/PlantUtilities.cc:366]: Ugh.  Yes.  This is silly.  :upside_down_face:

File: src/EnergyPlus/PlantUtilities.cc [hunk 2]
```diff
@@ -391,8 +392,9 @@ namespace PlantUtilities {
                     Node(OutletNode).MassFlowRate = CompFlow;
                     Node(InletNode).MassFlowRate = Node(OutletNode).MassFlowRate;
                     for (CompNum = 1; CompNum <= loop_side.Branch(BranchIndex).TotalComponents; ++CompNum) {
-                        CompInletNodeNum = comp.NodeNumIn;
-                        CompOutletNodeNum = comp.NodeNumOut;
+                        auto &thisComp(loop_side.Branch(BranchIndex).Comp(CompNum));
+                        CompInletNodeNum = thisComp.NodeNumIn;
+                        CompOutletNodeNum = thisComp.NodeNumOut;
```
Comment (Myoldmopar) [inline src/EnergyPlus/PlantUtilities.cc:397]: But quite better now.

**Review by Myoldmopar — APPROVED** [review]

**Comment by Myoldmopar** [comment]
I deleted the Mac results to see if the unit test continues to fail.

**Comment by Myoldmopar** [comment]
The unit test failure is now gone, so it's either fixed or at least not directly related to this change.  There are diffs showing in the latest Marvin results but they are due to a separate PR being merged in and this out of date.  I'm not going to let this linger running CI again.  Merging now.

**Pull request merged**